packages/context/test/unit/context.ts

```diff
@@ -63,10 +63,39 @@ describe('Context', () => {
     it('throws a helpful error when the binding is async', () => {
       ctx.bind('foo').toDynamicValue(() => Promise.resolve('bar'));
       expect(() => ctx.getSync('foo')).to.throw(/foo.*async/);
-   });
+    });
+  });
+
+  describe('context.bind().toProvider()', () => {
+    it('binds a provider class to a context', () => {
+      ctx.bind('foo').toProvider(MyProvider);
+      expect(ctx.contains('foo')).to.be.true();
+      expect(ctx.getBinding('foo').valueConstructor.name).to.be.equal('MyProvider');
```

Comment:
This is not correct. `valueConstructor` should contain the constructor used to instantiate the final value returned by the binding. Not `MyProvider`, but the constructor of whatever the `MyProvider.instance.value()` function returns.